pkg/utils/utils_test.go

```diff
@@ -0,0 +1,54 @@
+package utils
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestChanFromSlice(t *testing.T) {
+	// requireReceive is a helper function to check if a value can immediately be received from a channel.
+	requireReceive := func(t *testing.T, ch <-chan int, expected int) {
+		t.Helper()
+
+		select {
+		case v, ok := <-ch:
+			require.True(t, ok, "receiving should return a value")
+			require.Equal(t, expected, v)
+		default:
+			require.Fail(t, "receiving should not block")
+		}
+	}
+
+	// requireReceive is a helper function to check if the channel is closed and empty.
+	requireClosedEmpty := func(t *testing.T, ch <-chan int) {
+		t.Helper()
+
+		select {
+		case _, ok := <-ch:
+			require.False(t, ok, "receiving from channel should not return anything")
+		default:
+			require.Fail(t, "receiving should not block")
+		}
+	}
```

Comment:
These are closures, not regular functions. But why? What are they capturing?

Reply:
> What are they capturing?

Nothing.

> But why?

Initially scoping (i.e. aren't used elsewhere) but probably looks nicer the other way.

Reply:
> probably looks nicer the other way.

And now you _could_ make them generic. Just for fun. :)